sgpt/role.py

```diff
@@ -62,7 +62,10 @@ def __init__(
     @classmethod
     def create_defaults(cls) -> None:
         cls.storage.parent.mkdir(parents=True, exist_ok=True)
-        variables = {"shell": cls._shell_name(), "os": cls._os_name()}
+        if cfg.get("IN_CONTAINER") != "true":
+            variables = {"shell": cls._shell_name(), "os": cls._os_name()}
```

Comment:
I think it would be better to change `_os_name` and `_shell_name` functions, e.g.:
```python
@classmethod
def _shell_name(cls) -> str:
    overwrite_name = cfg.get("OVERWRITE_SHELL_NAME")
    if overwrite_name != "default":
        return overwrite_name

    current_platform = platform.system()
    if current_platform in ("Windows", "nt"):
        is_powershell = len(getenv("PSModulePath", "").split(pathsep)) >= 3
        return "powershell.exe" if is_powershell else "cmd.exe"
    return basename(getenv("SHELL", "/bin/sh"))
```